publish_jenkins_console.py

```diff
@@ -63,12 +63,25 @@ def _update_builds(console_log):
         _name, _id = _job.split(' #')
         _id = int(_id)
         if _name in JENKINS.keys():
-            ALL_BUILDS[_name] = _id
+            ALL_BUILDS[_name].add(_id)
             _build = _get_build(_name, _id)
-            BUILD_STATUS[_name] = _get_status(_build)
             _update_builds(_build.get_console())
 
 
+def _update_build_statuses():
+    for _name, _ids in ALL_BUILDS.items():
+        BUILD_STATUS[_name] = 'SUCCESS'
+        for _id in _ids:
+            _build = _get_build(_name, _id)
+            _status = _get_status(_build)
+            if _status == 'FAILURE':
+                BUILD_STATUS[_name] = _status
+            if _status == 'UNSTABLE' and BUILD_STATUS[_name] != 'FAILURE':
+                BUILD_STATUS[_name] = _status
+            if _status == 'ABORTED' and BUILD_STATUS[_name] != 'FAILURE':
+                BUILD_STATUS[_name] = _status
+
```

Comment:
I wanted to prevent possible overwritting of  FAILURE status by following concurrent jobs with UNSTABLE or ABORTED status since FAILURE status is more significant.
Without that it would always keep the status based on the latest job status which is not SUCCESS

Reply:
@FilipB in this case you may rewrite as simple,

```python
if _status == 'FAILURE':
  BUILD_STATUS[_name] = _status
  break
elif _status != 'SUCCESS':
  BUILD_STATUS[_name] = _status
```